src/cargo/ops/resolve.rs

```diff
@@ -832,7 +832,8 @@ fn register_patch_entries(
         // previous resolve graph, which is primarily what's done here to
         // build the `registrations` list.
         let mut registrations = Vec::new();
-        for dep in patches {
+        for dep_with_location in patches {
+            let (dep, _) = dep_with_location;
```

Comment:
I think I'd prefer destructuring the tuple in the `for` and then re-creating it in the `push`